File: ezpacking/src/main/java/org/zerock/ezpacking/domain/entity/Item.java
package org.zerock.ezpacking.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "items")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String category;

    @Column(columnDefinition = "TEXT")
    private String originalText;

    @Column(nullable = false)
    private boolean isAllowed;

    @Column(nullable = false)
    private boolean isConditional;


    @Column
    private String restrictions;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/item/entity/Item.kt
package com.example.apiezpz.item.entity

import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Entity
@Table(name = "items")
@EntityListeners(AuditingEntityListener::class)
data class Item(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    var name: String = "",

    @Column(nullable = false)
    var category: String = "",

    @Column(columnDefinition = "TEXT")
    var originalText: String? = null,

    @Column(nullable = false)
    var isAllowed: Boolean = false,

    @Column(nullable = false)
    var isConditional: Boolean = false,

    @Column
    var restrictions: String? = null,

    @CreatedDate
    var createdAt: LocalDateTime? = null,

    var createdDate: LocalDateTime? = null,
    
    @LastModifiedDate
    var updatedAt: LocalDateTime? = null
) 